fuse_examples/classification/MG_CMMD/dataset_folds.py

```diff
@@ -0,0 +1,208 @@
+                                            input_processors=input_processors,
+                                            gt_processors=gt_processors,
+                                            post_processing_func=post_cache_processing_func,
+                                            augmentor=None,
+                                            visualizer=visualiser)
+
+    test_data_source = FuseDataSourceFolds(input_source_gt,None,'test','ID1',[target],False,[4],5,partition_file_path)
+
+    test_dataset =  FuseDataSourceFolds(input_source=input_source_gt,
+                                            input_df=None,
+                                            phase='test',
+                                            no_mixture_id='ID1',
+                                            balance_keys=[target],
+                                            reset_partition_file=False,
+                                            folds=[4],
+                                            num_folds=5,
+                                            partition_file_name=partition_file_path)
+
+    lgr.info(f'- Load and cache data:')
+    validation_dataset.create(
+        pool_type='thread')  # use ThreadPool to create this dataset, to avoid cv2 problems in multithreading
+    lgr.info(f'- Load and cache data: Done')
+
+    return train_dataset, validation_dataset, test_dataset
+
+
+def merge_clinical_data_with_dicom_tags(root_path: str, target: str) -> str:
+    """
+    Creates a csv file that contains label for each image ( instead of patient as in dataset given file)
+    by reading metadata ( breast side and view ) from the dicom files and merging it with the input csv
+    If the csv already exists , it will skip the creation proccess
+    :param root_path    :       dataset root path
+    :return: the new csv file path
+    """
+    input_source = root_path + '/CMMD_clinicaldata_revision.csv'
+    combined_file_path = root_path + '/files_combined.csv'
+    if os.path.isfile(combined_file_path):
+        return combined_file_path
+    clinical_data = pd.read_csv(input_source)
+    scans = []
+    for patient in os.listdir(root_path + 'CMMD/'):
+        path = root_path + 'CMMD/' + patient
+        for dicom_file in glob.glob(os.path.join(path, '**/*.dcm'), recursive=True):
+            file = os.path.join(path, dicom_file)
+            dcm = pydicom.dcmread(file)
+            scans.append({'ID1': patient, 'LeftRight': dcm[0x00200062].value, 'file': file,
+                          'view': dcm[0x00540220].value.pop(0)[0x00080104].value})
+    dicom_tags = pd.DataFrame(scans)
+    merged_clinical_data = pd.merge(clinical_data, dicom_tags, how='outer', on=['ID1', 'LeftRight'])
+    merged_clinical_data = merged_clinical_data[merged_clinical_data[target].notna()]
+    merged_clinical_data.to_csv(combined_file_path)
+    return combined_file_path
+
+
+if __name__ == "__main__":
```

Comment:
Lets also remove all the unused mains. It's a bit confusing (we should remove it from old files as well)

Reply:
done